server/src/main/java/org/elasticsearch/health/node/selection/HealthNodeTaskExecutor.java

```diff
@@ -0,0 +1,194 @@
+import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.RemoteTransportException;
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+import org.elasticsearch.xcontent.ParseField;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.elasticsearch.health.node.selection.HealthNode.TASK_NAME;
+
+/**
+ * Persistent task executor that is managing the {@link HealthNode}.
+ */
+public final class HealthNodeTaskExecutor extends PersistentTasksExecutor<HealthNodeTaskParams> {
+
+    private static final Logger logger = LogManager.getLogger(HealthNodeTaskExecutor.class);
+
+    public static final Setting<Boolean> ENABLED_SETTING = Setting.boolSetting(
+        "health.node.enabled",
+        true,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    private final ClusterService clusterService;
+    private final PersistentTasksService persistentTasksService;
+    private final AtomicReference<HealthNode> currentTask = new AtomicReference<>();
+    private final ClusterStateListener taskStarter;
+    private final ClusterStateListener shutdownListener;
+    private volatile boolean enabled;
+
+    public HealthNodeTaskExecutor(
+        ClusterService clusterService,
+        PersistentTasksService persistentTasksService,
+        Settings settings,
+        ClusterSettings clusterSettings
+    ) {
+        super(TASK_NAME, ThreadPool.Names.MANAGEMENT);
+        this.clusterService = clusterService;
+        this.persistentTasksService = persistentTasksService;
+        this.taskStarter = this::startTask;
+        this.shutdownListener = this::shuttingDown;
+        this.enabled = ENABLED_SETTING.get(settings);
+        if (this.enabled) {
+            clusterService.addListener(taskStarter);
+            clusterService.addListener(shutdownListener);
+        }
+        clusterSettings.addSettingsUpdateConsumer(ENABLED_SETTING, this::enable);
+    }
+
+    private void enable(boolean enabled) {
+        this.enabled = enabled;
+        if (enabled) {
+            clusterService.addListener(taskStarter);
+            clusterService.addListener(shutdownListener);
+        } else {
+            clusterService.removeListener(taskStarter);
+            clusterService.removeListener(shutdownListener);
+            abortTaskIfApplicable("disabling health node via '" + ENABLED_SETTING.getKey() + "'");
+        }
+    }
+
+    @Override
+    protected void nodeOperation(AllocatedPersistentTask task, HealthNodeTaskParams params, PersistentTaskState state) {
+        HealthNode healthNode = (HealthNode) task;
+        currentTask.set(healthNode);
+        DiscoveryNode node = clusterService.localNode();
+        logger.info("Node {{{}{}}} is selected as the current health node.", node.getName(), node.getId());
```

Comment:
nit: is this `{{{}{}}}` syntax valid ? I believe we usually use `[{}]` to enclose values

Reply:
You are right of course! Fixed